src/wallet/api/schemas/walletApi.ts

```diff
@@ -0,0 +1,508 @@
+      melt: z.boolean(),
+    }),
+  }),
+});
+
+/**
+ * Schema for token balance information.
+ * Represents both unlocked and locked balances for a token.
+ */
+export const balanceSchema = z.object({
+  unlocked: bigIntCoercibleSchema,
+  locked: bigIntCoercibleSchema,
+});
+
+/**
+ * Schema for token authority balances.
+ * Represents mint and melt authority balances in both unlocked and locked states.
+ */
+export const authorityBalanceSchema = z.object({
+  unlocked: z.object({
+    mint: z.boolean(),
+    melt: z.boolean(),
+  }),
+  locked: z.object({
+    mint: z.boolean(),
+    melt: z.boolean(),
+  }),
+});
+
+/**
+ * Schema for balance object.
+ * Contains token info, balance, authorities, and transaction count.
+ */
+export const getBalanceObjectSchema = z.object({
+  token: tokenInfoSchema,
+  balance: balanceSchema,
+  tokenAuthorities: authorityBalanceSchema,
+  transactions: z.number(),
+  lockExpires: z.number().nullable(),
+});
+
+/**
+ * Response schema for token balances.
+ * Contains an array of balance objects for each token.
+ */
+export const balanceResponseSchema = baseResponseSchema.extend({
+  balances: z.array(getBalanceObjectSchema),
+});
+
+/**
+ * Schema for transaction proposal inputs.
+ * Represents the inputs that will be used in a transaction.
+ */
+export const txProposalInputsSchema = z.object({
+  txId: z.string(),
+  index: z.number(),
+  addressPath: AddressPathSchema,
+});
+
+/**
+ * Schema for transaction proposal outputs.
+ * Represents the outputs that will be created in a transaction.
+ */
+export const txProposalOutputsSchema = z.object({
+  address: z.string(),
+  value: bigIntCoercibleSchema,
+  token: z.string(),
+  timelock: z.number().nullable(),
+});
+
+/**
+ * Response schema for creating a transaction proposal.
+ * Contains the proposal ID and the transaction details.
+ */
+export const txProposalCreateResponseSchema = baseResponseSchema.extend({
+  txProposalId: z.string(),
+  inputs: z.array(txProposalInputsSchema),
+});
+
+/**
+ * Response schema for updating a transaction proposal.
+ * Contains the proposal ID and the transaction hex.
+ */
+export const txProposalUpdateResponseSchema = baseResponseSchema.extend({
+  txProposalId: z.string(),
+  txHex: z.string(),
+});
+
+/**
+ * Schema for full node version data.
+ * Contains network parameters and configuration values.
+ * Uses passthrough() to allow additional fields in the response without breaking validation,
+ * as the full node may add new fields in future versions without changing the API version.
+ */
+export const fullNodeVersionDataSchema = z
+  .object({
+    timestamp: z.number(),
+    version: z.string(),
+    network: z.string(),
+    minWeight: z.number(),
+    minTxWeight: z.number(),
+    minTxWeightCoefficient: z.number(),
+    minTxWeightK: z.number(),
+    tokenDepositPercentage: z.number(),
+    rewardSpendMinBlocks: z.number(),
+    maxNumberInputs: z.number(),
+    maxNumberOutputs: z.number(),
```

Comment:
Does it make sense to add as optional fields the genesis hashes and decimal places that will be available in the next full node version?

Reply:
I don't see why not!

Done